src/demographic_estimator.py

```diff
@@ -4,43 +4,45 @@
 """
 from deepface import DeepFace
 
-
-
 class Estimator:
     def __init__(self):
         pass
 
     def analyze_image(self, image_path):
         """
-        Takes an image path (Path object or string) and returns demographic info.
+        Takes an image path and returns a LIST of demographic info for all faces found.
         """
+        processed_faces = []
+        
         try:
-            # DeepFace requires a string.
             img_str = str(image_path)
+            # print(f"Analyzing: {img_str} ...") 
             
-            print(f"Analyzing: {img_str} ...")
-            
+            # 1. Analizi yap
             results = DeepFace.analyze(
                 img_path=img_str, 
                 actions=['age', 'gender', 'race'],
                 detector_backend='opencv', 
-                enforce_detection=False, # in case no face found.
+                enforce_detection=False, 
                 silent=True
             )
-            # DeepFace might return list or dict. 
-            if isinstance(results, list):
-                result = results[0]
-            else:
-                result = results
-            summary = {
-                "file": image_path.name,
-                "age": result['age'],
-                "gender": result['dominant_gender'],
-                "race": result['dominant_race'],
-                "confidence": result['face_confidence'] 
-            }
-            return summary
+            
+            if not isinstance(results, list):
+                results = [results]
+            for face in results:
+                if 'dominant_race' not in face:
+                    continue
+
+                summary = {
+                    "file": image_path.name,
+                    "age": face['age'],
+                    "gender": face['dominant_gender'],
+                    "race": face['dominant_race'],
+                    "confidence": face.get('face_confidence', 0)
+                }
+                processed_faces.append(summary)
+
+            return processed_faces
 
         except Exception as e:
```

Comment:
just cathing any Exception and returning an empty list in this case is the worst you can do. What Errors are you expecting? Can you recover from those with some simple steps? If not they should not be in your except block, because the info n what went wrong will just go missing.